lib/matplotlib/ticker.py

```diff
@@ -1170,25 +1170,181 @@ class LogitFormatter(Formatter):
+            parameters to change the default behavior.
+
+        minor_threshold : int (default: 25)
+            maximum number of locs for labelling some minor ticks. This
+            parameter have no effect if minor is False.
+
+        minor_number : int (default: 4)
+            number of ticks which are labelled when the number of ticks is
+            below the threshold.
+        """
+        self._use_overline = use_overline
+        self._minor = minor
+        self._labelled = set()
+        self._minor_threshold = minor_threshold
+        self._minor_number = minor_number
+
+    def use_overline(self, use_overline):
+        r"""
+        Switch display mode with overline for labelling p>1/2
+
+        Parameters
+        ----------
+        use_overline : bool (default: False)
+            if x > 1/2, with x = 1-v, indicate if x should be displayed as
+            $\overline{v}$. The default is to display $1-v$.
+        """
+        self._use_overline = use_overline
+
+    def label_minor(self, minor):
+        """
+        Define the formatting for labelling minor ticks.
+
+        Parameters
+        ----------
+        minor : bool
+            indicate if the formatter is formatting minor ticks or not.
+            Basically minor ticks are not labelled, except when only few ticks
+            are provided, the most espaced ticks are labelled. See others
+            parameters to change the default behavior.
+        """
+        self._minor = minor
+
+    def set_minor_threshold(self, minor_threshold):
+        """
+        Set the threshold for labelling minors ticks
+
+        Parameters
+        ----------
+        minor_threshold : int
+            maximum number of locs for labelling some minor ticks. This
+            parameter have no effect if minor is False.
+        """
+        self._minor_threshold = minor_threshold
+
+    def set_minor_number(self, minor_number):
+        """
+        Set the number of minor ticks to label when some minor ticks are
+        labelled.
+
+        Parameters
+        ----------
+        minor_number : int
+            number of ticks which are labelled when the number of ticks is
+            below the threshold.
+        """
+        self._minor_number = minor_number
+
+    def set_locs(self, locs):
+        self.locs = locs
+        self._labelled.clear()
+
+        if not self._minor:
+            return None
+        if all(
+            is_decade(x, tol=1e-7)
+            or is_decade(1 - x, tol=1e-7)
+            or (is_close_to_int(2 * x) and int(np.round(2 * x)) == 1)
+            for x in locs
+        ):
+            # minor ticks are subsample from ideal, so no label
+            return None
+        if len(locs) < self._minor_threshold:
+            if len(locs) < self._minor_number:
+                self._labelled.update(locs)
             else:
-                s = '$1-{:.5f}$'.format(1-x)
+                alocs = np.array(locs)
```

Comment:
This block is a bit obscure and could use a comment explaining what's happening.

Reply:
Done. I do not mark this thread as solve. It depends if you think it is clearer now.

Reply:
It's not clear what criterion (or even diff) below refers to.  In particular, is 0.1 really 1/base, or is it just some small value intended for tiebreaking? (in which case you could use tuple ordering instead?)

Reply:
It was only a tiebreak. I change it to tuple ordering, and I change the variable name to highlight the logic. Same as previous, I keep this thread open.